appdaemontestframework/hass_mocks.py

```diff
@@ -9,47 +10,21 @@
 CURRENT_APPDAEMON_VERSION = Version(appdaemon.utils.__version__)
 
 
-def is_appdaemon_version_at_least(version_as_string):
-    expected_appdaemon_version = Version(version_as_string)
-    return CURRENT_APPDAEMON_VERSION >= expected_appdaemon_version
-
-
-class _DeprecatedAppdaemonVersionWarning:
-    already_warned_during_this_test_session = False
-    min_supported_appdaemon_version = '4.0.0'
-
-    @classmethod
-    def show_warning_only_once(cls):
-        if cls.already_warned_during_this_test_session is True:
-            return
-        cls.already_warned_during_this_test_session = True
-
-        appdaemon_version_supported = is_appdaemon_version_at_least(
-                cls.min_supported_appdaemon_version
-        )
-        if not appdaemon_version_supported:
-            warnings.warn(
-                    "Appdaemon-Test-Framework will only support Appdaemon >={} "
-                    "in the next major release. "
-                    "Your current Appdemon version is {}".format(
-                            cls.min_supported_appdaemon_version,
-                            CURRENT_APPDAEMON_VERSION
-                    ),
-                    DeprecationWarning)
-
-
 class HassMocks:
     def __init__(self):
-        _DeprecatedAppdaemonVersionWarning.show_warning_only_once()
```

Comment:
Instead of completely removing the version checking logic, could you still check the version and if it’s v3 or below throw a very informative error message inviting the user to update to appdaemon version 4?

Otherwise we’ll end up with a “None extra argument thing” error like we had before we fixed support for v4

Reply:
Oh the distinction `warning` / `exception` is an excellent idea 😍